src/components/calcite-tooltip/calcite-tooltip.tsx

```diff
@@ -0,0 +1,308 @@
+import {
+  Component,
+  Element,
+  Host,
+  Method,
+  Prop,
+  State,
+  Watch,
+  h
+} from "@stencil/core";
+import { CSS, TooltipInteraction, TooltipPlacement } from "./resources";
+import Popper from "popper.js";
+import { VNode } from "@stencil/state-tunnel/dist/types/stencil.core";
+import { x16 } from "@esri/calcite-ui-icons";
+import CalciteIcon from "../../utils/CalciteIcon";
+
+@Component({
+  tag: "calcite-tooltip",
+  styleUrl: "calcite-tooltip.scss",
+  shadow: true
+})
+export class CalciteTooltip {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * Image source URL used to display an image above the text.
+   */
+  @Prop({ reflect: true }) image: string;
+
+  /**
+   * Image label.
+   */
+  @Prop({ reflect: true }) imageLabel: string;
+
+  /**
+   * Defines the way the user will interact with the tooltip.
+   * 'click' - Displays the tooltip on first click and hides on second click. Also provides the user with a close button within the tooltip.
+   * 'hover' - Displays the tooltip on mousover and hides on mouseout and displays the tooltip on focus and hides the tooltip on blur.
+   */
+  @Prop({ reflect: true }) interaction: TooltipInteraction = "hover";
+
+  @Watch("interaction")
+  interactionElementHandler(
+    newValue: TooltipInteraction,
+    oldValue: TooltipInteraction
+  ) {
+    this.removeReferenceListeners(oldValue);
+    this.addReferenceListeners(newValue);
+  }
+
+  /**
+   * Display and position the component.
+   */
+  @Prop({ reflect: true }) open = false;
+
+  @Watch("open")
+  openHandler(open: boolean) {
+    if (open) {
+      this.reposition();
+    } else {
+      this.destroyPopper();
+    }
+  }
+
+  /**
+   * Determines where the component will be positioned relative to the referenceElement.
+   */
+  @Prop({ reflect: true }) placement: TooltipPlacement = "auto";
+
+  @Watch("placement")
+  placementHandler() {
+    this.reposition();
+  }
+
+  /**
+   * Reference HTMLElement used to position this component.
+   */
+  @Prop() referenceElement: HTMLElement | string;
+
+  @Watch("referenceElement")
+  referenceElementHandler() {
+    this.removeReferenceListeners();
+    this._referenceElement = this.getReferenceElement();
+    this.addReferenceListeners();
+    this.destroyPopper();
+    this.reposition();
+  }
+
+  /**
+   * Tooltip text value to display.
+   */
+  @Prop({ reflect: true }) text: string;
```

Comment:
Can this be slotted content instead of a combination of properties (`text` & `image`)? It would be more flexible to allow users to pass either text or custom HTML to display in a tooltip, no?

Reply:
@paulcpederson @patrickarlt @julio8a I think this might be a question for you guys.

I was thinking if users want custom content, that would be the use for the `calcite-popover`. Whereas the `calcite-tooltip` is more restrictive.

Also, if we allow users to slot text and image, organizing it like this may be difficult.

![image](https://user-images.githubusercontent.com/1231455/66225057-5eb15280-e68c-11e9-99c3-878f98068280.png)

We could have two custom slots though, one for the image and one for the text.

Reply:
Added a slot for the image. Let me know how to proceed with the text.

Reply:
Not sure when or how the popover came into play. Can you give some details on its purpose

Reply:
> Not sure when or how the popover came into play. Can you give some details on its purpose

Maybe we can discuss this in a meeting?

Generally, tooltips are simpler and don't allow custom content. They are usually activated by hover only.

Generally, popovers are for more complex content and are activated via a click only.

We should meet and decide where to draw the line.

https://getbootstrap.com/docs/4.0/components/tooltips/
https://getbootstrap.com/docs/4.0/components/popovers/

@paulcpederson @patrickarlt @julio8a would love to get your feedback on how to proceed. Maybe tooltips should automatically work on hover and limit to text only. Popover should automatically work on click and allow custom content.